docs/cli-compile.md

```diff
@@ -11,3 +11,22 @@ It provides the following options:
 The `-n` or `--name` option provides control over the name of the top level
 kernel function generated to wrap the `decl`s. The default kernel name is
 `kernel`.
+
+## Output file
+
+The `-o` or `--out` option tells the compiler to place the output in a new
+file `<out>` instead of printing to STDOUT.
+
+## Backend
+
+The `-b` or `--backend` allows changing the backend used for generating code.
+Support backends are:
+
+1. `vivado`: Generates code compatible with Vivado HLS.
+2. `c++`: Generate code that can be compiled into an executable. See [this](cpp-runnable.md).
+
+## `run`
+
+The subcommand `run` requires the `-o` to be present. It automatically generates
+a CPP file and links it with the json data parser. See [this](cpp-runnable.md)
```

Comment:
More silly writing nitpicking: C++, not CPP; and JSON is capitalized. Finally, "this" is rarely all that helpful as text for a link—I recommend:

    See ["Generating Executables"](cpp-runnable.md) for details.

Also, it occurs to me that the default mode also generates a C++ file. So this might not be the clearest way to describe what the purpose of this mode is. Maybe a better way to introduce it would be to say that this is a mode for testing of these programs that lets you build a "standalone" executable by including a `main` function that lets you run on JSON data files.